### PR TITLE
Update configurations for the project

### PR DESCRIPTION
**Pull request by Vkt0r**
Hi guys,

Just a small changes in code. Big changes nevertheless updating the Pods. 

## CHANGES:

• Remove redundancy in the .gitignore regarding Carthage.
• Update the dependencies in the Cartfile & Podfile
• Update DemoTest to Swift 4 updating the project settings and removing warnings.
• Replace the deprecated `addDisposableTo(_:)` in favor of `disposed(by:)` in the DemoTest.

### CODE REVIEW

**Comment by Vkt0r** [comment]
Not sure about the current structure of the `CHANGELOG.md`. Please let me know about the structure because I'm seeing 

Current master
--------------

**Comment by ashfurrow** [comment]
@Vkt0r you should replace the `- Nothing yet!` with a markdown list item describing the change, similar to list items shown other versions.

**Comment by Vkt0r** [comment]
@ashfurrow Thanks 👍. Done !!

File: Demo/Demo/AppDelegate.swift [hunk 1]
```diff
@@ -14,30 +14,30 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
 
 
-    func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
+    private func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
```
Comment (ashfurrow) [inline Demo/Demo/AppDelegate.swift]: Any reason this is `private`?

**Review by ashfurrow — COMMENTED** [review]
Looks good, just one question.

**Comment by Vkt0r** [comment]
@ashfurrow 😅 Good catch!!, no idea where that was coming. It was XCode with the fix warnings, crazy.

**Review by ashfurrow — APPROVED** [review]

**Comment by ashfurrow** [comment]
I released this as 4.3.0, it should be available through Carthage and CocoaPods. Thanks again!

**Pull request merged**